proposals/0000-compiler-settings.md

```diff
@@ -0,0 +1,284 @@
+* Requiring code to use [Strict Memory Safety]().
+
+Since these compiler and language behaviors can only be manipulated via the
+command line and the compiler compiles one module at a time, users are forced to
+apply these flags one module at a time instead of one file at a time. This
+results in several forms of developer friction that we outline below.
+
+### Unnecessary Module Splitting
+
+Since compiler flags can only be applied to entire modules, one cannot apply a
+compiler flag to a subset of files of a module. When confronted with this, users
+are forced to split the subset of files into a separate module creating an
+unnecessary module split. We foresee that this will occur more often in the
+future if new features like [Controlling Default Actor Isolation]() and [Strict
+MemorySafety]() are accepted by Swift Evolution. Consider the following
+situations where unnecessary module splitting would be forced:
+
+* Consider a framework that by default uses `nonisolated` isolation, but wants
+  to expose a set of UI elements for users of the framework. In such a case, it
+  is reasonable to expect the author to want to make those specific files
+  containing UI elements to have a default isolation of `@MainActor` instead of
+  nonisolated. Without this feature one must introduce an additional module to
+  contain the UI elements for the framework when semantically there is really
+  only one framework.
+
+* Imagine an app that has enabled `@MainActor` isolation by default since it
+  contains mostly UI elements but that wishes to implement helper abstractions
+  that interact with a database on a background task. This helper code is
+  naturally expressed as having a default isolation of nonisolated implying that
+  one must either split the database helpers into a separate module or must mark
+  most declarations in the database code with nonisolated by hand.
+
+* Visualize a banking framework that contains both UI elements and functionality
+  for directly manipulating the account data of a customer. In such a case the
+  code that actually modifies customer data may want to enable strict memory
+  safety to increase security. In contrast, strict memory safety is not
+  necessary for the UI elements that the framework defines. To support both uses
+  cases, a module split must be introduced.
+
+### Preventing per-file based migration of large modules using warningsAsErrors
+
+Swift has taken the position that updating modules for a new language mode or
+feature should be accomplished by:
+
+1. Enabling warnings on the entire module by enabling upcoming features from the
+   command line.
+
+2. Updating the module incrementally until all of the warnings have been
+   eliminated from the module.
+
+3. Updating the language version of the module so from that point on warnings
+   will become errors to prevent backsliding.
+
+This creates developer friction when applied to large modules since the size of
```

Comment:
A general editorial comment: there are a number of long sentences in this proposal that would benefit from commas. For example, in the above, you probably want to insert a comma before "since":

> This creates developer friction when applied to large modules, since the size of the module makes it difficult to eliminate all of the warnings at once.